### PR TITLE
Deduping redirects

### PR DESCRIPTION
**Pull request by nicoolas25**
Hello!

This PR addresses issue #3050: it prevents two different possible redirection issues in the `source/_redirects.htaccess`.

I added some consistency checks while the redirect files are being parsed in order to check:

- that we don't have twice the same alias going to a different destination, and
- that we don't have an alias that is the destination of another redirecting rule (or the opposite).

A second commit is present to fix some consistency issue I found. I'm not so sure about that commit, please let me know!

@ksigler7

<details>
<summary>Screenshots</summary>

screenshot from 2017-10-14 16-41-47

screenshot from 2017-10-14 16-41-56

</details>

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/docs?pullRequest=3078) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/docs?pullRequest=3078) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/docs?pullRequest=3078) it.</sub>

File: source/_redirects.htaccess [hunk 1]
```diff
@@ -250,7 +250,6 @@ Redirect 301 /Classroom/Basics/does_sendgrid_support_end_to_end_tls.html http://
 Redirect 301 /Classroom/Basics/email_flow.html http://sendgrid.com/docs/Classroom/Basics/Email_Infrastructure/email_flow.html
 Redirect 301 /Classroom/Basics/everything_about_dmarc.html http://sendgrid.com/docs/Classroom/Basics/Email_Infrastructure/everything_about_dmarc.html
 Redirect 301 /Classroom/Basics/link_shorteners.html http://sendgrid.com/docs/Classroom/Build/Add_Content/link_shorteners.html
-Redirect 301 /Classroom/Basics/password.html http://sendgrid.com/docs/Classroom/Basics/Security/password.html
```
Comment (nicoolas25) [inline source/_redirects.htaccess]: This was duplicated, I think keeping the `https` version make sense.

File: source/_redirects.htaccess [hunk 2]
```diff
@@ -606,7 +605,6 @@ Redirect 301 /User_Guide/Email_Deliverability/Undelivered_Email/spam.html https:
 Redirect 301 /User_Guide/Email_Settings/categories.html https://sendgrid.com/docs/Glossary/categories.html
 Redirect 301 /User_Guide/managing_unsubscribes.html https://sendgrid.com/docs/User_Guide/Suppressions/index.html
 Redirect 301 /User_Guide/Marketing_Campaigns/api.html https://sendgrid.com/docs/API_Reference/Web_API_v3/Marketing_Campaigns/contactdb.html
-Redirect 301 /User_Guide/Marketing_Campaigns/campaigns.html https://sendgrid.com/docs/API_Reference/Web_API_v3/Marketing_Campaigns/index.html
```
Comment (nicoolas25) [inline source/_redirects.htaccess]: This was duplicated, I kept the last one (the one being currently active).

**Comment by ksigler7** [comment]
@nicoolas25 this is so epic - thanks so much for adding this in!!! 
I threw some test cases at this, and it is working really well!

Your second pull request is completely on point, that was left over from a test a few months ago, and obviously not removed at the time.

**Comment by thinkingserious** [comment]
Hello @nicoolas25,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
@nicoolas25 you rock!

**Comment by nicoolas25** [comment]
> We want to show our appreciation by sending you some swag. Could you please fill out this form so we can send it to you?

Of course, thank you to make the issue clear an approachable with labels!

**Pull request merged**